system/density/buffers/buffer.c

```diff
@@ -0,0 +1,184 @@
+/*
```

Comment:
We already use git submodule with other 3rd part libraries, but not all I think.
@richardkchapman - should we for these?